pybop/parameters/priors.py

```diff
@@ -120,6 +244,10 @@ def sigma(self):
         """
         return self.scale
 
+    @property
+    def n_parameters(self):
```

Comment:
Can the number of parameters be different from 1? I think we should keep to individual parameter priors for now. In which case, let's move the `n_parameters` property to the joint prior only.